virttest/qemu_devices/qdevices.py

```diff
@@ -3957,3 +3957,60 @@ def _cmdline_raw(self):
             # -machine allows empty line
             return ""
         return super()._cmdline_raw()
+
+
+class QCSSBus(QSparseBus):
+    """
+    CSS Bus representation (bus&cssid&ssid&devno, uses hex digits)
+    """
+
+    def __init__(self, busid, bus_type, aobject):
+        """bus&cssid&ssid&devno, 255 cssid,4 ssid and 65536 device numbers"""
+        super(QCSSBus, self).__init__(
+            "bus",
+            [["cssid", "ssid", "devno"], [255, 4, 65536]],
+            busid,
+            bus_type,
+            aobject,
+        )
+
+    @staticmethod
+    def _addr2stor(addr):
+        """
+        Converts internal addr to storable/hashable address
+        :param addr: internal address [addr1, addr2, ...]
+        :return: storable address "addr1-addr2-..."
+        """
+        out = ""
+        for value in addr:
+            if value is None:
+                out += "*-"
+            else:
+                out += "%s-" % value
+        if out:
+            return out[:-1]
+        else:
+            return "*"
+
+    def _dev2addr(self, device):
+        """Read the values in base of 16 (hex)"""
+        addr = device.get_param("addr")
+        if isinstance(addr, int):  # only devno
+            return [254, 0, addr]
```

Comment:
All values should be from `cssid` `ssid`, not hard code, although `range(cssid))[-1]`, `range(ssid)[0]`, I don't know what's the correct value here.

Reply:
`device.get_param("devno")`

Reply:
I could confirm that on s390x we could use fe.0.xxxx for testing by default, @smitterl do you have any other opinion?

Reply:
@huth perhaps you could help us to have a look, if we make cssid and ssid static and only dynamic for the devno

Reply:
@fbq815 , yes, I think hard-coding to fe.0.xxxx should be fine, that's also the recommendation in https://gitlab.com/qemu-project/qemu/-/blob/master/docs/system/s390x/css.rst ... unless you want to test more than 64k devices in one test, but that's rather unlikely I think, isn't it?

Reply:
@huth yes, then we just use the hard-coding of fe.0.xxxx it is, thanks for helping
@maxujun please updated the code and modify it as what we discussed before.